ansible/roles/ci/doc/best_practices.md

```diff
@@ -0,0 +1,92 @@
+# Best Practices
+
+## Splitting work between different CI servers
+
+Currently build tools support three hosted CI solutions 
+
+  * [Shipppable](https://www.shippable.com)
+  * [Semaphore](https://semaphoreci.com)
+  * [Circle](https://circleci.com)
+ 
+Also it supports local run which might be used by any CI setup on the local servers or for manual execution in Docker container.
+
+The one of the possible approaches is to split checks between different servers to make builds faster and to reduce cost for private repositories.
+E.g. Circle and Shippable currently propose one worker for private and public repositories.
+So it can be used to run fast checkup for all repositories such as roslint code style checks.
+
+Also module build_pr_only can be used to build only Pull Requests in order to save server time for other repositories.
+
+Semaphore has scheduled build feature which can be used to run heavy repository checks durng the night.
+
+## Open source repositories
+
+Open source repositories can be run free of charge on any of these server  
+
+## Local server
+
+In case if you want to setup Docker based CI server on your local servers you can use as following
```

Comment:
If you want to setup Docker based CI server on your local servers you can do the following: